ch360/documents.go

```diff
@@ -77,29 +96,29 @@ type classifyDocumentResponse struct {
 	} `json:"classification_results"`
 }
 
-func (client *DocumentsClient) Create(ctx context.Context, fileContents []byte) (string, error) {
-	body := bytes.NewBuffer(fileContents)
-	response, err := newRequest(ctx, "POST", client.baseUrl+"/documents", body).
+func (client *DocumentsClient) Create(ctx context.Context, fileContents io.Reader) (Document,
+	error) {
+	response, err := newRequest(ctx, "POST", client.baseUrl+"/documents", fileContents).
 		issue(client.requestSender)
 
 	if err != nil {
-		return "", err
+		return Document{}, err
```

Comment:
Should we return `nil` here, and similarly lower down?

Reply:
I went back and forth a bit on this - the method returns a value (as opposed to a pointer), and so as it stands it needs to return a `Document` (and can't be `nil`). I did that to match the `GetAll` method, which returns a `DocumentList` (or `[]Document`, as opposed to a `[]*Document`).